kratos/solving_strategies/strategies/residualbased_linear_strategy.h

```diff
@@ -358,6 +358,15 @@ class ResidualBasedLinearStrategy
 
         this->GetScheme()->Predict(BaseType::GetModelPart(), r_dof_set, rA, rDx, rb);
 
+        if(BaseType::GetModelPart().MasterSlaveConstraints().size() != 0)
+        {
+            TSparseSpace::SetToZero(rDx);
+            p_scheme->Update(BaseType::GetModelPart(), r_dof_set, rA, rDx, rb);
```

Comment:
shouldn't these two line here should be reversed .. first apply m-s constraints and then update ??

Reply:
yes this should be reversed. I did correct this in the NR  but not here. Marking as todo for tomorrow.